app/src/org/commcare/models/database/app/AppDatabaseUpgrader.java

```diff
@@ -200,6 +217,131 @@ private boolean upgradeSevenEight(SQLiteDatabase db) {
         }
     }
 
+    // Migrate records form FormProvider and InstanceProvider to new FormDefRecord and InstanceRecord respectively
+    private boolean upgradeEightNine(SQLiteDatabase db) {
+        db.beginTransaction();
+        android.database.Cursor cursor = null;
```

Comment:
Strictly speaking no lines of code should occur after db.begin until the try starts.